lib/transformers/DockerCompose.js

```diff
@@ -286,8 +286,34 @@ DockerComposeTransformer.prototype.validateIntrusiveFeatures = function(yamlStri
     }
 
     try {
-        var compositionObject = YAML.load(yamlString);
+        var compositionObject = YAML.safeLoad(yamlString);
+
+        // remove all services that were marked by the engine explicitly to not be validated
+        if (_.get(compositionObject, 'version', '').toString() === '2') {
```

Comment:
Instead of duplicating the logic, perhaps we can do:
```
const services;
if (object.version is 2) {
services = object.services 
}
else {
services = object
}

... filter ...
```